### PR TITLE
maint(ct): clean up contract tests 1

### PR DESCRIPTION
**Pull request by smartcontracts**
**Description**
First commit in a series of commits to clean up the contract tests. Made
a deploy helper function to remove some ugly nested awaits. Also
generally starting to remove unnecessary Factory objects.

### CODE REVIEW

File: packages/contracts/test/helpers/utils/deploy.ts [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+
+export const deploy = async (
+  name: string,
+  opts?: {
+    args?: any[]
+    signer?: any
+  }
+) => {
+  const factory = await hre.ethers.getContractFactory(name, opts?.signer)
+  return factory.deploy(...(opts?.args || []))
```
[the remark this answers is not in the document]
Comment (tynes) [reply packages/contracts/test/helpers/utils/deploy.ts:11]: typescript type enforcement makes this function pretty easy to use and lets us get rid of needing to import all of the factories in our code. The `getContractFactory` helper is meant to be used in code that interacts with the actual networks if I understand correctly?

[the remark this answers is not in the document]
Comment (smartcontracts) [reply packages/contracts/test/helpers/utils/deploy.ts:11]: Yeah `getContractFactory` is really only used externally. This helper is mainly meant to reduce the amount of duplicated code. Just slightly cuts down the tests, nice to not have to create a factory every time. And yeah I want to apply this same change to all of the code at the same time, but I felt like the PR would be way too big and sit in review forever if I did that.

File: packages/contracts/test/contracts/L1/deployment/ChugSplashDictator.spec..ts [hunk 2]
```diff
@@ -1,51 +1,40 @@
-/* External Imports */
```
Comment (maurelian) [inline packages/contracts/test/contracts/L1/deployment/ChugSplashDictator.spec..ts:1]: What's up with the `..ts` file extension?

File: packages/contracts/test/helpers/utils/deploy.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (maurelian) [reply packages/contracts/test/helpers/utils/deploy.ts:11]: > Yeah `getContractFactory` is really only used externally

In that case it should really have a docstring on it both for external users, and to clarify the intent to internal contributors.

File: packages/contracts/test/contracts/L1/deployment/ChugSplashDictator.spec..ts (hunk 2 above)
Comment (smartcontracts) [reply packages/contracts/test/contracts/L1/deployment/ChugSplashDictator.spec..ts:1]: Lol, surprised we never noticed that.

**Comment by maurelian** [comment]
Will approve once the filename is fixed.

**Comment by smartcontracts** [comment]
> Will approve once the filename is fixed.

Fixed

**Review by mslipper — APPROVED** [review]

**Review by maurelian — APPROVED** [review]
LGTM

**Pull request merged**